plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -774,10 +780,28 @@ private function get_final_query_args( $common_query_values, $query, $is_exclude
 			$applied_filters_query,
 			$date_query,
 			$price_query_args,
-			$handpicked_query
+			$handpicked_query,
+			$collection_query
 		);
 	}
 
+	/**
+	 * Get any collection-specific query args to merge.
+	 *
+	 * @param string $collection  The name of the collection.
+	 * @param array  $query       Query from block context.
+	 *
+	 * @return array The collection-specific query to merge.
+	 */
+	private function get_core_collection_query( $collection, $query ) {
+		$collection = preg_match( '/^woocommerce\/product-collection\/(.*)/', $collection, $matches ) ? $matches[1] : '';
```

Comment:
Loving this approach! It helps illustrate the impact of each collection on the query!
One suggestion: Instead of `preg_match` I'd use `strpos`?

As per PHP [docs](https://www.php.net/manual/en/function.preg-match.php):
```
Do not use preg_match() if you only want to check if one string is contained in another string. Use [strpos()](https://www.php.net/manual/en/function.strpos.php) instead as it will be faster.
````

Reply:
I was lazy so I used `preg_match()` to get the collection name without bothering to check the front of the string and remove the first n characters 😅 I pushed up a new approach that I'm _super_ happy with though so this is gone lol